aws/resource_aws_emr_instance_fleet.go

```diff
@@ -0,0 +1,587 @@
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     instanceTypeConfigSchema(),
+			},
+			"launch_specifications": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     launchSpecificationsSchema(),
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"target_on_demand_capacity": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"target_spot_capacity": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"provisioned_on_demand_capacity": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"provisioned_spot_capacity": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func instanceTypeConfigSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"bid_price": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Required: false,
+				ForceNew: true,
+			},
+			"bid_price_as_percentage_of_on_demand_price": {
+				Type:     schema.TypeFloat,
+				Optional: true,
+				ForceNew: true,
+			},
+			"configurations": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     configurationSchema(),
+			},
+			"ebs_optimized": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ebs_config": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     ebsConfigurationSchema(),
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"weighted_capacity": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  1,
+			},
+		},
+	}
+}
+
+func launchSpecificationsSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"spot_specification": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				MinItems: 1,
+				Elem:     spotSpecificationSchema(),
+			},
+		},
+	}
+}
+
+func spotSpecificationSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"block_duration_minutes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  0,
+			},
+			"timeout_action": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAwsEmrSpotProvisioningTimeOutAction,
+			},
+			"timeout_duration_minutes": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+	}
+}
+
+func configurationSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"classification": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			//"configurations": {
```

Comment:
Did it resolve in a loop issue or why did you comment out these lines?

Reply:
yes! checking the documentation here we can define configurations inside configurations but when I defined it, it resolves in a loop issue trying to "compile" the schema. Any idea on how to fix this? Because I was looking in the code and I didn't find any alternative/solution.

Reply:
The doc is not really clear, but it seems that no real recursion is required here. Limiting the nesting to one level should be enough. So, you could use a new schema (e.g. additionalConfigurationSchema) that only has a "classification" and "properties" attribute.